ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Server_Configurations.py

```diff
@@ -66,29 +74,78 @@ def parse_yaml(hosts, config):
             }
 
             for provider in providers:
```

Comment:
This whole function is a nest pyramid of doom :-).  Is there a way to extract some of the logic to smaller private functions and use some sort of filtering operation to avoid using so many nested loops?

Reply:
Probably, but this will need some work and testing, I'll log this as in improvement in the Nagios Enhancements EPIC. and revisit it as part of that work, if that's ok with you? ( This issue is tracking the future improvements.  #2981